src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2980,6 +2980,18 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
                 return Conversion.Identity;
             }
 
+            if (argument.Kind == BoundKind.DefaultPendingInference)
+            {
+                if (parameterType.IsStructType() && !parameterType.IsNullableType())
+                {
+                    return Conversion.Identity;
+                }
+                else
+                {
+                    return Conversion.NoConversion;
+                }
+            }
+
```

Comment:
This should not be necessary; it should be a consequence of ClassifyImplicitConversionFromExpression, below.